src/utilities/common.ts

```diff
@@ -213,11 +223,16 @@ export function modelFromFieldConfigs (fieldConfigs: IFieldConfigPartial[], data
         const { field, nullify } = fieldConfig
           , formValue = get(data, field)
           , shouldNullify = nullify && !formValue && formValue !== false
-          , value = shouldNullify ? null : formValue
-          ;
+          , value = isTypeObjectSearchCreate(fieldConfig) && isObject(formValue) && !has(formValue, ID_ATTR)
+            ? modelFromFieldConfigs(fieldConfig.createFields, formValue)
+            : shouldNullify ? null : formValue
+            ;
 
         set(returnValues, field, value);
       });
 
+    const id = get(data, ID_ATTR);
```

Comment:
Why are these lines being added? If the `id` was to be in the model, wouldn't it be a field?

Reply:
It's funny, yeah, it would seem that way. But if we stopped including ids from model unless they had a ( hidden? ) field defined, most of CoOp and registry would stop working. This is something we've done since the original V2 FormManager. Most patch functions patch to `/${data.id}` without ever thinking about how they never included an id in their field sets. It's just one of those things you always expect to be there.